src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -581,5 +592,80 @@ impl<'index, I: Iterator<Item = RSIndexResult<'index>>> Iterator for FilterMaskR
     }
 }
 
+/// A reader that filters out records that do not match a given numeric filter. It is used to
+/// filter records in an index based on their numeric value, allowing only those that match the
+/// specified filter to be returned.
+///
+/// This should only be wrapped around readers that return numeric records.
```

Comment:
Could we enforce this within the type system somehow?

Reply:
For now reducing this to predicates won't be possible because the index reader will need to return the numeric filter

https://github.com/RediSearch/RediSearch/blob/aa1241c5da5c08be956f8bdc90c5b2a4f348d0be/src/inverted_index/inverted_index.h#L237-L240